### PR TITLE
[Doc] Update documentation regarding authentication with Azure-managed Service Principal using GITHUB OIDC

### PR DESCRIPTION
**Pull request by parthban-db**
## Changes

Update documentation regarding authentication with Azure-managed Service Principal using GITHUB OIDC.

## Tests

- [ ] `make test` run locally
- [x] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -314,6 +314,36 @@ resource "databricks_user" "my-user" {
...
+  azure_workspace_resource_id = azurerm_databricks_workspace.this.id
+  azure_client_id             = var.client_id
+  azure_tenant_id             = var.tenant_id
+}
+
+resource "databricks_user" "my-user" {
+  user_name = "test-user@databricks.com"
+}
+```
+
```
Comment (alexott) [inline docs/index.md:346]: No, we need to move it below after the description of the attributes for client secret auth. And then describe how it should be setup in Github, and what attributes are arequired

**Review by alexott — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 1 above)
Comment (parthban-db) [reply docs/index.md:346]: Do I add the description of the attributes again?

File: docs/index.md [hunk 2]
```diff
@@ -326,6 +326,38 @@ There are `ARM_*` environment variables provide a way to share authentication co
...
+Follow the [Configuring OpenID Connect in Azure](https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-azure). You can then use this Azure service principal to authenticate in databricks.
```
Comment (alexott) [inline docs/index.md:359]: Can we list required and optional parameters for it? It's ok to copy relevant parts from the previous section

File: docs/index.md [hunk 3]
```diff
@@ -326,6 +326,49 @@ There are `ARM_*` environment variables provide a way to share authentication co
...
+  user_name = "test-user@databricks.com"
+}
+```
+
+Follow the [Configuring OpenID Connect in Azure](https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-azure). You can then use the Azure service principal to authenticate in databricks. 
+
+* `azure_workspace_resource_id` - (optional) `id` attribute of [azurerm_databricks_workspace](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/databricks_workspace) resource. Combination of subscription id, resource group name, and workspace name. Required with `azure_use_msi` or `azure_client_secret`.
+
+* `azure_client_id` - (optional) This is the Azure Enterprise Application (Service principal) client id. This service principal requires contributor access to your Azure Databricks deployment. Alternatively, you can provide this value as an environment variable `ARM_CLIENT_ID`.
+* `azure_tenant_id` - (optional) This is the Azure Active Directory Tenant id in which the Enterprise Application (Service Principal)
```
Comment (alexott) [inline docs/index.md:364]: Are they really optional for this auth method? Same for `auth_type` ...

**Review by tanmay-db — APPROVED** [review]
LGTM as the comments have been addressed.

**Review by alexott — APPROVED** [review]

**Pull request merged**